SurfaceTopography/Uniform/Imputation.py

```diff
@@ -0,0 +1,234 @@
+#
+# Copyright 2021 Lars Pastewka
+#
+# ### MIT license
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+"""
+Filter pipelines for data imputation (filling undefined data points)
+"""
+
+import numpy as np
+import scipy.sparse
+
+from _SurfaceTopography import assign_patch_numbers as assign_patch_numbers_area
+
+from ..HeightContainer import UniformTopographyInterface
+from ..UniformLineScanAndTopography import DecoratedUniformTopography
+
+# Stencils for determining nearest-neighbor relationships on a square grid
+nn_stencil = [(1, 0), (0, 1), (-1, 0), (0, -1)]
+nnn_stencil = [(1, 1), (1, 0), (1, -1), (0, -1), (-1, -1), (-1, 0), (-1, 1), (0, 1)]
+
+
+def coordination(c, periodic, stencil=nn_stencil):
+    """
+    Return a map with coordination numbers, i.e. number of neighboring patches that also contact
```

Comment:
Docstring: is this an old code from contactmechanics ? Maybe be more explicit and say the coordination number is computed from the number of neighboring  patches where  c is true

Reply:
Yes this was contact mechanics specific. I've updated the documentation string.